### PR TITLE
Notify patients in select states in IHCI

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** [sc-9556](https://app.shortcut.com/simpledotorg/story/9556/send-reminders-only-in-ap-tn-and-wb-in-ihci)

## Because

Restrict notifications to a select list of states in IHCI.

## This addresses

Adds a whitelist of states in India that will be included in criteria for eligibility in experiments.

### CODE REVIEW

File: app/models/experimentation/notifications_experiment.rb [hunk 1]
```diff
@@ -54,6 +55,14 @@ def self.exclude_bangladesh_blocks(patients)
         .where.not(block_region: {id: excluded_block_ids.presence})
     end
 
+    def self.restrict_indian_states(patients)
```
Comment (olttwa) [inline app/models/experimentation/notifications_experiment.rb:58]: Rename the method to `exclude_non_experiment_indian_states`?
`exclude` seems better than `restrict`

**Review by olttwa — CHANGES_REQUESTED** [review]

File: app/models/experimentation/notifications_experiment.rb (hunk 1 above)
Comment (prabhanshuguptagit) [reply app/models/experimentation/notifications_experiment.rb:58]: wdyt of `include_selected_indian_states`? It applies a whitelist so it's not actually `exclude`ing.

Comment (olttwa) [reply app/models/experimentation/notifications_experiment.rb:58]: What this function is doing is removing non_experiment states from the list.
I'll prefer `filter_non_experiment_indian_states` to `include_experiment_indian_states`. Your call. No more discussion on this.

Comment (harimohanraj89) [reply app/models/experimentation/notifications_experiment.rb:58]: I like `exclude_india_states`. Consistent with the exclusion above (exclude_bangladesh_blocks). Note that it's `India`, not `Indian`, to be consistent 😅

Comment (prabhanshuguptagit) [reply app/models/experimentation/notifications_experiment.rb:58]: >What this function is doing is removing non_experiment states from the list.
There is no notion of `non_experiment` states, only `included` states.

Going with `exclude_india_states`, can't think of a name that captures the whitelist well.

Comment (prabhanshuguptagit) [reply app/models/experimentation/notifications_experiment.rb:58]: Actually, `include_only_allowed_india_states` fits the bill. Picking this.

**Pull request merged**